src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionsArray = [];
+  const resultObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(resultObject, action.extraData, actionsArray);
+        break;
+      case 'removeProperties':
+        removeProperties(resultObject, action.keysToRemove, actionsArray);
+        break;
+      case 'clear':
+        clear(actionsArray, resultObject);
+        break;
+    }
+  }
+
+  return actionsArray;
+}
+
+function clear(actionsArray, resultObject) {
+  for (const value in resultObject) {
+    delete resultObject[value];
+  }
+  actionsArray.push({ ...resultObject });
```

Comment:
The `clear` function should not push the `resultObject` after clearing it. It should push a new empty object to the `actionsArray`.